Khan/wonder-blocks
# feat(dropdown): SingleSelect with search filter

## Summary

Similar to the MultiSelect component (https://github.com/Khan/wonder-blocks/pull/479), these changes introduce the `isFilterable` prop to `<SingleSelect />`.

This allows us to filter options inside the single select component.

### Test plan
1. Open https://5e1bf4b385e3fb0020b7073c-vojuppvbfz.chromatic.com/?path=/story/dropdown-singleselect--custom-styles
2. Verify that the search input is present and that we can filter options by query.

![single-filter](https://user-images.githubusercontent.com/843075/82601821-252b4a00-9b7e-11ea-970f-d0941253d318.gif)
